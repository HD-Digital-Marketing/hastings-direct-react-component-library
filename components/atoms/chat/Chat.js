import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import Icon from '../icon/Icon'
import './chat.scss'
import ChatIcon from '../../../src/img/icons/chat.svg'

const Chat = props => {

    const {
        isSmall = false,
        className = '',
        children
    } = props

    const chatClass = classNames('a-chat', className, {
        'a-chat--small': isSmall
    })

    return(
        <div className={chatClass}>
            <span className="a-chat__label">{children}</span> <Icon svgWidth={25} svgHeight={25} svgFill="#fff" svgImage={ChatIcon} />
        </div>
    )
}

Chat.propTypes = {
    isSmall: PropTypes.bool,
    className: PropTypes.string,
    children: PropTypes.string
}

export default Chat
