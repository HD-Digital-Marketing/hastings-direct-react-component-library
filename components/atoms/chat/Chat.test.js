import React from 'react'
import renderer from 'react-test-renderer'
import Chat from './Chat'

describe('Check the chat renders', () => {
    test('it should match the snapshot', () => {

        const tree = renderer
            .create(
                <Chat isSmall={true}>Chat</Chat>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()

    })
})