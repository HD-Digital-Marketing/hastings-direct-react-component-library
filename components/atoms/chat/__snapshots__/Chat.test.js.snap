// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the chat renders it should match the snapshot 1`] = `
<div
  className="a-chat a-chat--small"
>
  <span
    className="a-chat__label"
  >
    Chat
  </span>
   
  <span
    className="a-icon"
    style={
      Object {
        "backgroundColor": undefined,
        "borderColor": undefined,
        "borderWidth": 2.5,
        "height": 25,
        "width": 25,
      }
    }
  />
</div>
`;
